oop/src/PurchaseOrders/infrastructure/repositories/PurchaseOrder.repo.ts

```diff
@@ -1,15 +1,30 @@
 import { UUID } from "../../../utilities/uuid";
 import { IPORepository } from "../../domain/PurchaseOrder.repo-interface";
 import { PurchaseOrder } from "../../domain/PurchaseOrder.entity";
+import { PONumber } from "../../domain/PONumber.value-object";
+
+const byPONumber = (a: PurchaseOrder, b: PurchaseOrder) => {
+  return (a?.number?.digits ?? 0) > (b?.number?.digits ?? 0) ? -1 : 1;
+};
 
 export class PORepository implements IPORepository {
   purchaseOrders: PurchaseOrder[] = [];
   async save(po: PurchaseOrder) {
```

Comment:
Taking some liberties with our js repo. The po number value object is stored as is where as in a real db repo this would need to be mapped to the string value.